### PR TITLE
Add Serverless External Registry job

### PR DESCRIPTION
**Pull request by michal-hudy**
**Description**

Changes proposed in this pull request:

- Added tests that verifies it external registry works

**Related issue(s)**

kyma-project/kyma#8204

### CODE REVIEW

**Comment by michal-hudy** [comment]
Docker images will be removed as a part of this issue: https://github.com/kyma-project/test-infra/issues/2530

File: prow/scripts/cluster-integration/kyma-gke-external-registry.sh [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+"${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/get-helm-certs.sh"
+
+shout "Test Kyma"
+date
+KYMA_TESTS="serverless serverless-long core-test-external-solution" "${TEST_INFRA_SOURCES_DIR}/prow/scripts/kyma-testing.sh"
+
+shout "Verify if internal docker registry is disabled"
+date
+verify_internal_registry
```
Comment (magicmatatjahu) [inline prow/scripts/cluster-integration/kyma-gke-external-registry.sh:324]: Maybe check that before testing? :)

File: prow/scripts/cluster-integration/kyma-gke-external-registry.sh [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+    --data "gateways.istio-ingressgateway.loadBalancerIP=${GATEWAY_IP_ADDRESS}" \
+    --label "component=istio"
+
+dockerPassword=/tmp/kyma-gke-integration/dockerPassword.json
+mkdir -p /tmp/kyma-gke-integration
+< "${GCR_PUSH_GOOGLE_APPLICATION_CREDENTIALS}" tr -d '\n' > /tmp/kyma-gke-integration/dockerPassword.json
+"${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/create-secret.sh" --name "serverless-external-registry-overrides" \
+    --data "dockerRegistry.enableInternal=false" \
+    --data "dockerRegistry.username=_json_key" \
+    --file "dockerRegistry.password=${dockerPassword}" \
```
Comment (magicmatatjahu) [inline prow/scripts/cluster-integration/kyma-gke-external-registry.sh:286]:
```suggestion
    --file "dockerRegistry.jsonKey=${dockerPassword}" \
```

File: prow/scripts/cluster-integration/kyma-gke-external-registry.sh [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+    --data "gateways.istio-ingressgateway.loadBalancerIP=${GATEWAY_IP_ADDRESS}" \
+    --label "component=istio"
+
+dockerPassword=/tmp/kyma-gke-integration/dockerPassword.json
+mkdir -p /tmp/kyma-gke-integration
+< "${GCR_PUSH_GOOGLE_APPLICATION_CREDENTIALS}" tr -d '\n' > /tmp/kyma-gke-integration/dockerPassword.json
+"${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/create-secret.sh" --name "serverless-external-registry-overrides" \
+    --data "dockerRegistry.enableInternal=false" \
+    --data "dockerRegistry.username=_json_key" \
+    --file "dockerRegistry.jsonKey=${dockerPassword}" \
```
Comment (magicmatatjahu) [inline prow/scripts/cluster-integration/kyma-gke-external-registry.sh:286]:
```suggestion
    --file "dockerRegistry.password=${dockerPassword}" \
```

File: prow/jobs/kyma/kyma-gke-external-registry.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      # following regexp won't start build if only Markdown files were changed
+      run_if_changed: "^((resources\\\\serverless\\S+|installation\\S+|tools/kyma-installer\\S+)(\\.[^.][^.][^.]+$|\\.[^.][^dD]$|\\.[^mM][^.]$|\\.[^.]$|/[^.]+$))"
+      labels:
+        preset-build-pr: "true"
+        <<: *gke_job_labels_template
+      extra_refs:
+        - <<: *test_infra_ref
+          base_ref: master
+
+# postsubmits:
```
Comment (Ressetkk) [inline prow/jobs/kyma/kyma-gke-external-registry.yaml:61]: Will it be used in future? If not then please remove the commented code

File: templates/templates/kyma-gke-external-registry.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        preset-build-release: "true"
+        <<: *gke_job_labels_template
+      extra_refs:
+        - org: kyma-project
+          repo: test-infra
+          path_alias: github.com/kyma-project/test-infra
+          base_ref: release-{{ . }}
+{{- end }}
+
+# postsubmits:
```
Comment (Ressetkk) [inline templates/templates/kyma-gke-external-registry.yaml:75]: same as before

File: prow/jobs/kyma/kyma-gke-external-registry.yaml (hunk 4 above)
Comment (michal-hudy) [reply prow/jobs/kyma/kyma-gke-external-registry.yaml:61]: It will be uncommented when the functionality will be merged to master...

**Pull request merged**